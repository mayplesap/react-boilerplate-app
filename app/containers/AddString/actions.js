/*
 * AddString Actions
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 */

import {
  CHANGE_STRING,
  SUBMIT_STRING_SUCCESS,
  SUBMIT_STRING_ERROR,
} from './constants';

/**
 * Changes the input field of the form
 *
 * @return {object} An action object with a type of CHANGE_STRING
 */
export function changeString(input) {
  return {
    type: CHANGE_STRING,
    input,
  };
}

/**
 * Dispatched when the input is submited by the request saga
 *
 * @return {object}      An action object with a type of SUBMIT_STRING_SUCCESS passing the input
 */
export function stringSubmitted(submit) {
  return {
    type: SUBMIT_STRING_SUCCESS,
    submit,
  };
}

/**
 * Dispatched when submitting the input fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of SUBMIT_STRING_ERROR passing the error
 */
export function stringSubmitError(error) {
  return {
    type: SUBMIT_STRING_ERROR,
    error,
  };
}
