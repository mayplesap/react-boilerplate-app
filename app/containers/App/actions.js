/*
 * App Actions
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 */

import {
  LOAD_STRING,
  LOAD_STRING_SUCCESS,
  LOAD_STRING_ERROR,
} from './constants';

/**
 * Load the list, this action starts the request saga
 *
 * @return {object} An action object with a type of LOAD_STRING
 */
export function loadList() {
  return {
    type: LOAD_STRING,
  };
}

/**
 * Dispatched when the list is loaded by the request saga
 *
 * @return {object}      An action object with a type of LOAD_STRING_SUCCESS passing the repos
 */
export function listLoaded(strings) {
  return {
    type: LOAD_STRING_SUCCESS,
    strings,
  };
}

/**
 * Dispatched when loading the list fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of LOAD_STRING_ERROR passing the error
 */
export function listLoadingError(error) {
  return {
    type: LOAD_STRING_ERROR,
    error,
  };
}
