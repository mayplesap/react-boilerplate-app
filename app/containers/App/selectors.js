import { createSelector } from 'reselect';
import { initialState } from './reducer';

const selectGlobal = state => state.global || initialState;

const selectRouter = state => state.router;

const makeSelectList = () =>
  createSelector(
    selectGlobal,
    globalState => globalState.strings,
  );

const makeSelectLoading = () =>
  createSelector(
    selectGlobal,
    globalState => globalState.loading,
  );

const makeSelectError = () =>
  createSelector(
    selectGlobal,
    globalState => globalState.error,
  );

const makeSelectLocation = () =>
  createSelector(
    selectRouter,
    routerState => routerState.location,
  );

export {
  selectGlobal,
  makeSelectLocation,
  makeSelectList,
  makeSelectLoading,
  makeSelectError,
};
