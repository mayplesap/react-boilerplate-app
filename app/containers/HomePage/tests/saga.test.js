/**
 * Tests for HomePage sagas
 */

import { put, takeLatest } from 'redux-saga/effects';

import { LOAD_STRING } from 'containers/App/constants';
import { listLoaded, listLoadingError } from 'containers/App/actions';

import listData, { getList } from '../saga';

/* eslint-disable redux-saga/yield-effects */
describe('getList Saga', () => {
  let getListGenerator;

  // do all the stuff that happens beforehand automatically in the beforeEach
  beforeEach(() => {
    getListGenerator = getList();

    const selectDescriptor = getListGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();
  });

  it('should dispatch the listLoaded action if it requests the data successfully', () => {
    const response = ['hello', 'world', 'how'];
    const putDescriptor = getListGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(listLoaded(response)));
  });

  it('should call the listLoadingError action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = getListGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(listLoadingError(response)));
  });
});

describe('listData Saga', () => {
  const listDataSaga = listData();

  it('should start task to watch for LOAD_STRING action', () => {
    const takeLatestDescriptor = listDataSaga.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(LOAD_STRING, getList));
  });
});
