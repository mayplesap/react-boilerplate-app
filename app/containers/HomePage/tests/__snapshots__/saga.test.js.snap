// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getList Saga should call the listLoadingError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "http://localhost:3001/strings",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getList Saga should dispatch the listLoaded action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "http://localhost:3001/strings",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
